core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -282,7 +282,9 @@ private[spark] object Utils extends Logging {
         if (dir.exists() || !dir.mkdirs()) {
           dir = null
         } else {
-          if (!chmod700(dir)) {
+          // Restrict file permissions via chmod if available.
+          // For Windows this step is ignored.
+          if (!isWindows && !chmod700(dir)) {
```

Comment:
What if we swapped the order of these conditionals so that we first attempt to change the permissions and then ignore failures if we're running on Windows?  Would that work, or would it fail with an exception?